cmd/notation/internal/truststore/truststore.go

```diff
@@ -0,0 +1,226 @@
+	maxDepth := strings.Count(root, string(os.PathSeparator)) + depth
+
+	return filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() && strings.Count(path, string(os.PathSeparator)) > maxDepth {
+			return fs.SkipDir
+		}
+		info, err := d.Info()
+		if err != nil {
+			return err
+		}
+		if info.Mode().IsRegular() {
+			if _, err := corex509.ReadCertificateFile(path); err != nil {
+				return err
+			}
+			fmt.Println(path)
+		}
+		return nil
+	})
+}
+
+// ShowCerts writes out details of certificates
+func ShowCerts(certs []*x509.Certificate) {
+	fmt.Println("Display certificate details. Starting from leaf certificate if it's a certificate chain.")
+	fmt.Println("--------------------------------------------------------------------------------")
+	for ind, cert := range certs {
+		showCert(cert)
+		if ind != len(certs)-1 {
+			fmt.Println("--------------------------------------------------------------------------------")
+		}
+	}
+}
+
+// showCert displays details of a certificate
+func showCert(cert *x509.Certificate) {
+	fmt.Println("Issuer:", cert.Issuer)
+	fmt.Println("Subject:", cert.Subject)
+	fmt.Println("Valid from:", cert.NotBefore)
+	fmt.Println("Valid to:", cert.NotAfter)
+	fmt.Println("IsCA:", cert.IsCA)
+
+	h := sha1.Sum(cert.Raw)
+	fmt.Println("Thumbprints:", hex.EncodeToString(h[:]))
+}
+
+// DeleteAllCerts deletes all certificate files from the trust store
+// under dir truststore/x509/storeType/namedStore
+func DeleteAllCerts(storeType, namedStore string, confirmed bool, errorSlice []error) []error {
+	path, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore)
+	if err == nil {
+		truststorePath := fmt.Sprintf("truststore/x509/%s/%s", storeType, namedStore)
+		prompt := fmt.Sprintf("Are you sure you want to remove all certificate files under: %q?", truststorePath)
+		confirmed, err := cmdutil.AskForConfirmation(os.Stdin, prompt, confirmed)
+		if err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+			return errorSlice
+		}
+		if !confirmed {
+			return errorSlice
+		}
+
+		if err = os.RemoveAll(path); err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+		} else {
+			// write out on success
+			fmt.Printf("Successfully cleared %s\n", path)
+			return nil
+		}
+	} else {
+		errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+	}
+	return errorSlice
+}
+
+// DeleteCert deletes a specific certificate file from the
+// trust store, namely truststore/x509/storeType/namedStore/cert
+func DeleteCert(storeType, namedStore, cert string, confirmed bool, errorSlice []error) []error {
+	path, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore, cert)
+	if err == nil {
+		truststorePath := fmt.Sprintf("truststore/x509/%s/%s/%s", storeType, namedStore, cert)
```

Comment:
truststorePath is not a valid path for windows because widows uses backslashes while Mac and Linux uses slashes.

Reply:
Oh yeah, truststorePath is a concise path and only used for display purpose. In other words, it's only used during printing to the command line prompt. To operate on OS level, the correct system path in line 183 is used.